File: core/src/org/joltsphere/mechanics/ArenaContactListener.java
package org.joltsphere.mechanics;

import com.badlogic.gdx.physics.box2d.Contact;
import com.badlogic.gdx.physics.box2d.ContactImpulse;
import com.badlogic.gdx.physics.box2d.ContactListener;
import com.badlogic.gdx.physics.box2d.Fixture;
import com.badlogic.gdx.physics.box2d.Manifold;
import com.badlogic.gdx.utils.Array;

public class ArenaContactListener implements ContactListener {

	public Array<Integer> playerContacts;
	public byte pvpContact = 0;
	
	private Fixture fa, fb;
	
	public ArenaContactListener(int playerCount) {
		playerContacts = new Array<Integer>();
		for (int i = 1; i <= playerCount; i++) {
			playerContacts.add(0);
		}
	}
	
	public void beginContact(Contact contact) {
		
		fa = contact.getFixtureA();
		fb = contact.getFixtureB();
		
		for (int i = 1; i <= playerContacts.size; i++) {
			if (isContacting("ground", "p"+i)) playerContacts.set(i - 1, playerContacts.get(i - 1) + 1);
		}
		
		if (isContacting("p1", "p2")) pvpContact++;
	
	}
	
	public void endContact(Contact contact) {
		
		fa = contact.getFixtureA();
		fb = contact.getFixtureB();
		
		for (int i = 1; i <= playerContacts.size; i++) {
			if (isContacting("ground", "p"+i)) playerContacts.set(i - 1, playerContacts.get(i - 1) - 1);
		}
		
		if (isContacting("p1", "p2")) pvpContact--;
	
	}

	public void preSolve(Contact contact, Manifold oldManifold) {}

	public void postSolve(Contact contact, ContactImpulse impulse) {}
	
	private boolean isContacting(String f1, String f2) {
		if (fa.getUserData().equals(f1) && fb.getUserData().equals(f2))
			return true;
		else if (fa.getUserData().equals(f2) && fb.getUserData().equals(f1))
			return true;
		else return false;
	}
	
}

File: core/src/org/joltsphere/mechanics/ArenaContactListener.kt
package org.joltsphere.mechanics

import com.badlogic.gdx.physics.box2d.*
import com.badlogic.gdx.utils.Array

/* TODO
    * - Create pvp contacts array
    */
class ArenaContactListener(playerCount: Int) : ContactListener {

    var playerContacts: Array<Int> = Array<Int>()

    var pvpContact: Byte = 0

    private lateinit var bA: Body
    private lateinit var bB: Body

    init {
        for (i in 1..playerCount) {
            playerContacts.add(0)
        }
    }

    override fun beginContact(contact: Contact) {

        bA = contact.fixtureA.body
        bB = contact.fixtureB.body

        for (i in 1..playerContacts.size) {
            if (isContacting("ground", "p" + i)) playerContacts.set(i - 1, playerContacts.get(i - 1) + 1)
        }

        if (isContacting("p1", "p2")) pvpContact++

    }

    override fun endContact(contact: Contact) {

        bA = contact.fixtureA.body
        bB = contact.fixtureB.body

        for (i in 1..playerContacts.size) {
            if (isContacting("ground", "p" + i)) playerContacts.set(i - 1, playerContacts.get(i - 1) - 1)
        }

        if (isContacting("p1", "p2")) pvpContact--

    }

    override fun preSolve(contact: Contact, oldManifold: Manifold) {}

    override fun postSolve(contact: Contact, impulse: ContactImpulse) {}

    private fun isContacting(f1: String, f2: String): Boolean {
        if (bA.userData == f1 && bB.userData == f2)
            return true
        else if (bA.userData == f2 && bB.userData == f1)
            return true
        else
            return false
    }

}
